### PR TITLE
Feature/593 enhance scenario action

### PR DESCRIPTION
**Pull request by ismaelbej**

### CODE REVIEW

File: src/planwise/component/scenarios.clj [hunk 1]
```diff
@@ -36,13 +36,42 @@
...
+  [counters]
+  (letfn [(format-summary
+            ([] "")
+            ([x] x)
+            ([x y] (str x " and " y))
+            ([x y z] (str x ", " y " and " z)))
+          (build-action
+            [count label]
+            (if (> count 0) (str label " " (common/pluralize count "provider"))))]
+    (str/capitalize (apply format-summary (filter some? (map build-action counters action-labels))))))
```
Comment (ggiraldez) [inline src/planwise/component/scenarios.clj:64]: This probably reads better with the threading macro `->>`.
```
(->> (map build-action counters action-labels)
     (filter some?)
     (apply format-summary))
```

File: src/planwise/component/scenarios.clj [hunk 2]
```diff
@@ -36,13 +36,42 @@
...
+     (letfn [(action-to-index [action]
+               (case action
+                 "create-provider" 0
+                 "upgrade-provider" 1
+                 "increase-provider" 2))]
+       (update-in counters [(action-to-index (:action change))] inc)))
+   [0 0 0]
+   changeset))
+
+(def action-labels ["create" "upgrade" "increase capacity for"])
```
Comment (ggiraldez) [inline src/planwise/component/scenarios.clj:52]: If using `frequencies`, you'd need to change this to a map from `:action` into label. You will lose the ordering, but you can make it explicit in `build-summary` later.

File: src/planwise/component/scenarios.clj [hunk 3]
```diff
@@ -36,13 +36,42 @@
...
+          (build-action
+            [count label]
+            (if (> count 0) (str label " " (common/pluralize count "provider"))))]
+    (str/capitalize (apply format-summary (filter some? (map build-action counters action-labels))))))
+
 (defn- build-changeset-summary
   [changeset]
-  (let [providers (count (filter #(-> % :initial nil?) changeset))
+  (let [changeset (filter #(-> % :initial nil?) changeset)
+        counters (count-actions changeset)
```
Comment (ggiraldez) [inline src/planwise/component/scenarios.clj:69]: This would turn into `(frequencies (map :action changeset))` I think.

**Comment by ismaelbej** [comment]
@ggiraldez The proposed changes were applied.

**Pull request merged**